packages/flutter_tools/test/general.shard/android/android_studio_test.dart

```diff
@@ -914,6 +967,31 @@ void main() {
       ProcessManager: () => FakeProcessManager.any(),
     });
 
+    testUsingContext('when given an Android Studio newer than any known version, finds Java version by assuming latest known Android Studio version', () {
+      const String studioHomeFilePath =
+          '$homeLinux/.cache/Google/AndroidStudio99999.99.99/.home';
+      const String studioInstallPath = '$homeLinux/AndroidStudio';
+
+      fileSystem.file(studioHomeFilePath)
+        ..createSync(recursive: true)
+        ..writeAsStringSync(studioInstallPath);
+
+      fileSystem.directory(studioInstallPath).createSync();
+
+      final String expectedJdkLocationFor2022 = fileSystem.path.join(studioInstallPath, 'jbr', 'bin', 'java');
+      fileSystem.file(expectedJdkLocationFor2022).createSync(recursive: true);
+
+      final AndroidStudio studio = AndroidStudio.allInstalled().single;
+
+      expect(studio.version, greaterThanOrEqualTo(Version(99999, null, null)));
```

Comment:
why is this greater than or equal to?

Reply:
The idea was that we don't care what the exact version is since that's not the point of the test--more so it's to just to make sure that tool parsing at least _some_ version and that the version is very high.

But I suppose there's no good reason to _not_ exactly match on the version, so I've updated the tests to do so.